pubsub/pubsub_test.go

```diff
@@ -72,3 +121,89 @@ func TestSendReceive(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestLotsOfMessagesAndSubscriptions(t *testing.T) {
+	howManyToSend := int(1e3)
+	ctx := context.Background()
+	dt := &driverTopic{}
+
+	// Make subscriptions and start goroutines to receive from them.
+	var subs []*pubsub.Subscription
+	nSubs := 100
+	var wg sync.WaitGroup
+	for i := 0; i < nSubs; i++ {
+		ds := NewDriverSub()
+		dt.subs = append(dt.subs, ds)
+		s := pubsub.NewSubscription(ctx, ds, pubsub.SubscriptionOptions{})
+		subs = append(subs, s)
+		wg.Add(howManyToSend)
```

Comment:
1

Reply:
Substituting 1 gives this result:

```
~/src/github.com/google/go-cloud/pubsub$ go test
panic: sync: negative WaitGroup counter

goroutine 32 [running]:
sync.(*WaitGroup).Add(0xc0000c0070, 0xffffffffffffffff)
	/usr/lib/google-golang/src/sync/waitgroup.go:74 +0x137
```

Reply:
Okay, I see this works with your comment below. Good point. Changed it.